dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpServiceContextImpl.java

```diff
@@ -3032,24 +3030,25 @@ private void handleNettyPreload(String uri) {
                          + headers);
         }
 
-        ChannelFuture promise = handler.encoder().writePushPromise(nettyContext, currentStreamId, nextPromisedStreamId, headers, 0,
-                                                                   new VoidChannelPromise(this.nettyContext.channel(), true));
-
-        promise.addListener(future -> {
-            if (future.isSuccess()){
-
-            }
-            else {
-                future.cause().printStackTrace();
+        this.nettyContext.channel().eventLoop().execute(new Runnable() {
+            @Override
+            public void run() {
+                ChannelFuture promise = handler.encoder().writePushPromise(nettyContext, currentStreamId, nextPromisedStreamId, headers, 0,
+                                                                   new VoidChannelPromise(nettyContext.channel(), true));
+                promise.addListener(future -> {
+                    if (future.isSuccess()){
+                        // Should we process the new request here when we ensure we wrote out a push promise?
```

Comment:
Can we get a dashboard item to investigate/followup on this one?

Reply:
Created https://github.com/OpenLiberty/open-liberty/issues/31439